lib/Chat/ChatManager.php

```diff
@@ -315,7 +321,13 @@ public function getLastReadMessageFromLegacy(Room $chat, IUser $user): int {
 	}
 
 	public function getUnreadCount(Room $chat, int $lastReadMessage): int {
-		return $this->commentsManager->getNumberOfCommentsForObjectSinceComment('chat', (string) $chat->getId(), $lastReadMessage, 'comment');
+		$key = $chat->getId() . '-' . $lastReadMessage;
```

Comment:
add comment like:
"for a given message id $lastReadMessage we cache the number of messages that exist past that message, which happen to also be the number of unread messages, because this is expensive to query/compute."